pyccel/ast/literals.py

```diff
@@ -222,3 +222,54 @@ def get_default_literal_value(dtype):
     else:
         raise TypeError('Unknown type')
     return value
+
+#------------------------------------------------------------------------------
+
+def convert_to_literal(value, dtype = None, precision = None):
+    """ Convert a python value to a pyccel Literal
+
+    Parameters
+    ----------
+    value     : int/float/complex/bool/str
+                The python value
+    dtype     : DataType
+                The datatype of the python value
+                Default : Matches type of 'value'
+    precision : int
+                The precision of the value in the generated code
+                Default : python precision (see default_precision)
+    """
+    if dtype is None:
+        if isinstance(value, int):
+            dtype = NativeInteger()
+        elif isinstance(value, float):
+            dtype = NativeReal()
+        elif isinstance(value, complex):
+            dtype = NativeComplex()
+        elif isinstance(value, bool):
+            dtype = NativeBool()
+        elif isinstance(value, str):
+            dtype = NativeString()
+        else:
+            raise TypeError('Unknown type')
+
+    if precision is None and dtype is not NativeString():
+        precision = default_precision[str(dtype)]
+
+    if isinstance(dtype, NativeInteger):
+        value = LiteralInteger(value, precision)
+    elif isinstance(dtype, NativeReal):
+        value = LiteralFloat(value, precision)
+    elif isinstance(dtype, NativeComplex):
+        value = LiteralComplex(value.real, value.imag, precision)
+    elif isinstance(dtype, NativeBool):
+        if value:
+            value = LiteralTrue(precision)
+        else:
+            value = LiteralFalse(precision)
+    elif isinstance(dtype, NativeString):
+        value = LiteralString(value)
+    else:
+        raise TypeError('Unknown type')
+
+    return value
```

Comment:
Would you mind using another name for the function result? `value` is an input argument already.

Reply:
Fixed in e0f925c

Reply:
Thanks